src/export/InstanceExporter.ts

```diff
@@ -447,6 +455,73 @@ export class InstanceExporter implements Fishable {
+    instanceDef: InstanceDefinition,
+    instanceOfStructureDefinition: StructureDefinition
+  ) {
+    const nonChoiceSlicedElements = instanceOfStructureDefinition.elements.filter(
+      e => e.slicing && !e.path.endsWith('[x]')
+    );
+    for (const slicingEl of nonChoiceSlicedElements) {
+      // Use the sliced element's path to get all potential element values from the instance
+      const { values } = collectValuesAtElementIdOrPath(slicingEl.id, instanceDef);
+      // Find the values that do not have named slices
+      const unnamedValues = values.filter(v => v != null && v._sliceName == null);
+      if (unnamedValues.length) {
+        // Find the values that do have named slices (for comparison)
+        const namedValues = values
+          .filter(v => v?._sliceName != null)
+          .map(v => {
+            const cleanValue = cloneDeep(v);
+            delete cleanValue._sliceName;
+            return { name: v._sliceName, value: cleanValue };
+          });
+        if (namedValues.length) {
+          for (const value of unnamedValues) {
+            // It's a match if the unnamed value is a compatible superset of the named value (i.e., slice)
+            const matchedNamedValues = namedValues.filter(nv => isMatch(value, nv.value));
+            if (matchedNamedValues.length) {
+              const matchedNames = matchedNamedValues.map(nv => nv.name);
+              if (matchedNamedValues.every(nm => isEqual(value, nm.value))) {
+                logger.warn(
+                  `${instanceDef.id} has an array item that is exactly the same as a required slice, but does not ` +
+                    'use the slice name in its path. This may result in unexpected items in your array. Usually you ' +
+                    'can remove this item from your FSH, since FSH instances inherit required assigned values from ' +
+                    "their profile. Alternately, you can modify your FSH to use the slice name in the item's path. " +
+                    ' See: https://hl7.org/fhir/uv/shorthand/reference.html#sliced-array-paths\n' +
+                    `  Path:  ${slicingEl.path}\n` +
+                    `  Slice: ${matchedNames.join(' or ')}\n` +
+                    `  Value: ${JSON.stringify(value)}`,
+                  fshDef.sourceInfo
+                );
+              } else {
+                logger.warn(
+                  `${instanceDef.id} has an array item that matches a required slice, but does not use the slice name ` +
+                    'in its path. This may result in unexpected items in your array. If the item is intended to be ' +
+                    "in the slice, modify your FSH to use the slice name in the item's path. See: " +
+                    'https://hl7.org/fhir/uv/shorthand/reference.html#sliced-array-paths\n' +
+                    `  Path:  ${slicingEl.path}\n` +
+                    `  Slice: ${matchedNames.join(' or ')}\n` +
+                    `  Value: ${JSON.stringify(value)}`,
```

Comment:
I'm wondering if the `Value` should be included in a `debug` level log. There was a project I noticed that had this warning on `Bundle.entry` and the `Value` log is pretty dense because it's a large JSON object. Maybe the usefulness most times outweighs this particular case coming up, just was curious what you thought.

Reply:
That's a good point... I think the value is too useful to hide in a debug log, but perhaps we should truncate it after some number of characters... I'll think on that.

Reply:
OK. Commit ed89951 truncates values > 300 characters. The number 300 is kind of arbitrary, but it seems like a good compromise to me.  Let me know what you think. Here's an example from `nightingaleproject/vital_records_fhir_messaging_ig`:
<img width="992" alt="image" src="https://user-images.githubusercontent.com/2278253/216474150-26534cc7-47b2-48df-94f1-3079718e4686.png">

Reply:
The 300 value looks reasonable to me, so I like it! I think this will give enough info to point them to the right place without cluttering the output. Thanks!